providers/infrastructure-vsphere/v1.4.1/yttcc/overlay.yaml

```diff
@@ -95,36 +95,42 @@ spec:
       #@overlay/replace
       - class: tkg-worker-windows
         name: md-0
+        #@overlay/match missing_ok=True
+        #@ if data.values.CLUSTER_PLAN == "prodcc":
+        replicas: #@ data.values.WORKER_MACHINE_COUNT_0
+        #@ else:
         replicas: #@ data.values.WORKER_MACHINE_COUNT
+        #@ end
         #@overlay/match missing_ok=True
         metadata:
           annotations:
-            #! VVV TODO(tenczar) os-name handling
-            run.tanzu.vmware.com/resolve-os-image: image-type=ova,os-name=windows
+            run.tanzu.vmware.com/resolve-os-image: #@ "image-type=ova,os-name={}".format(data.values.OS_NAME)
+        #@ if data.values.VSPHERE_AZ_0:
+        #@overlay/match missing_ok=True
+        failureDomain: #@ data.values.VSPHERE_AZ_0
+        #@ end
       #@ if data.values.CLUSTER_PLAN == "prodcc":
       #@overlay/append
       - class: tkg-worker-windows
         name: md-1
         replicas: #@ data.values.WORKER_MACHINE_COUNT_1
-        #@ if data.values.VSPHERE_AZ_1:
-        failureDomain: #@ data.values.VSPHERE_AZ_1
-        #@ end
         #@overlay/match missing_ok=True
         metadata:
           annotations:
-            #! VVV TODO(tenczar) os-name handling
-            run.tanzu.vmware.com/resolve-os-image: image-type=ova,os-name=windows
+            run.tanzu.vmware.com/resolve-os-image: #@ "image-type=ova,os-name={}".format(data.values.OS_NAME)
```

Comment:
out of curiosity, what are the possible values of OS_NAME in the windows scenario? This all should be fine as long as we were already using these OS_NAMES in the legacy version.

Reply:
that's a fair point, we do not use OS_NAME for legacy. 
If we start to use the same OS_NAME for cp and Windows WL it will conflict, I will roll back to `os-name=windows` since this is the default OVF prop from image-builder.